src/main/java/xyz/nucleoid/plasmid/game/player/isolation/PlayerResetter.java

```diff
@@ -5,12 +5,7 @@
 import net.minecraft.server.network.ServerPlayerEntity;
 import net.minecraft.util.registry.Registry;
 
-public final class PlayerResetter {
-    private final NbtCompound resetNbt;
-
-    public PlayerResetter(NbtCompound resetNbt) {
-        this.resetNbt = resetNbt;
-    }
+public record PlayerResetter(NbtCompound resetNbt) {
```

Comment:
While classes like this can be a record, their internal values are not supposed to be accessible, and arguably should not be.